src/node_dotenv.cc

```diff
@@ -98,73 +110,46 @@ bool Dotenv::ParsePath(const std::string_view path) {
     result.append(buf.base, r);
   }
 
-  using std::string_view_literals::operator""sv;
-  auto lines = SplitString(result, "\n"sv);
+  // Convert line breaks to the same format
+  std::string lines = result;
+  std::regex_replace(lines, std::regex("\r\n?"), "\n");
 
-  for (const auto& line : lines) {
-    ParseLine(line);
-  }
-  return true;
-}
+  std::smatch match;
+  while (std::regex_search(lines, match, LINE)) {
+    const std::string key = match[1].str();
 
-void Dotenv::AssignNodeOptionsIfAvailable(std::string* node_options) {
-  auto match = store_.find("NODE_OPTIONS");
+    // Default undefined or null to an empty string
+    std::string value = match[2].str();
 
-  if (match != store_.end()) {
-    *node_options = match->second;
-  }
-}
+    // Remove whitespace
+    value = std::regex_replace(value, std::regex("^\\s+|\\s+$"), "");
 
-void Dotenv::ParseLine(const std::string_view line) {
-  auto equal_index = line.find('=');
+    // Check if double-quoted
+    const char maybeQuote = value[0];
 
-  if (equal_index == std::string_view::npos) {
-    return;
-  }
-
-  auto key = line.substr(0, equal_index);
+    // Remove surrounding quotes
+    value =
+        std::regex_replace(value, std::regex("^(['\"`])([\\s\\S]*)\\1$"), "$2");
 
-  // Remove leading and trailing space characters from key.
-  while (!key.empty() && std::isspace(key.front())) key.remove_prefix(1);
-  while (!key.empty() && std::isspace(key.back())) key.remove_suffix(1);
+    // Expand newlines if double quoted
+    if (maybeQuote == '"') {
+      value = std::regex_replace(value, std::regex("\\\\n"), "\n");
```

Comment:
I believe string erase and remove would be faster and more readable than regex replace. The same applies for the next line as well.

```
str.erase(remove(str.begin(), str.end(), ch), str.end());
```